src/main/scala/sigmastate/serialization/DataJsonEncoder.scala

```diff
@@ -86,34 +98,59 @@ object DataJsonEncoder {
       assert(len == tArr.length, s"Type $t doesn't correspond to value $arr")
       var obj = mutable.MutableList.empty[(String, Json)]
       cfor(0)(_ < len, _ + 1) { i =>
-        obj += (s"_${i + 1}" -> encodeData[SType](arr(i),tArr(i)))
+        obj += (s"_${i + 1}" -> encodeData[SType](arr(i), tArr(i)))
       }
       Json.fromFields(obj.toList)
+    case SGroupElement =>
+      val w = SigmaSerializer.startWriter()
+      DataSerializer.serialize(v, tpe, w)
+      encodeBytes(w.toBytes)
+    case SAvlTree =>
+      val w = SigmaSerializer.startWriter()
+      DataSerializer.serialize(v, tpe, w)
+      encodeBytes(w.toBytes)
+    case SSigmaProp =>
+      val w = SigmaSerializer.startWriter()
+      DataSerializer.serialize(v, tpe, w)
+      encodeBytes(w.toBytes)
+    case SBox =>
+      val ergoBox = boxToErgoBox(v.asInstanceOf[Box])
+      var obj = mutable.MutableList.empty[(String, Json)]
+      obj += ("value" -> encodeData(ergoBox.value.asInstanceOf[SType#WrappedType], SLong))
+      obj += ("tree" -> encodeBytes(ErgoTreeSerializer.DefaultSerializer.serializeErgoTree(ergoBox.ergoTree)))
+      obj += "tokens" -> encodeData(ergoBox.additionalTokens.map { case (id, amount) =>
+        (Colls.fromArray(id), amount)
+      }.asInstanceOf[SType#WrappedType], SCollectionType(STuple(SCollectionType(SByte), SLong)))
+      ergoBox.additionalRegisters.foreach { case (id, value) =>
+        obj += (s"r${id.number}" -> encode[SType](value.value, value.tpe))
+      }
+      obj += ("txId" -> encodeBytes(ergoBox.transactionId.toBytes))
+      obj += ("index" -> encodeData(ergoBox.index.asInstanceOf[SType#WrappedType], SShort))
```

Comment:
I suspect this type cast is not necessary.

Reply:
that leads to an error: found `Short`, required `T#WrappedType`